### PR TITLE
Removes SUPERMATTER_IGNORE_1 flag, some fixes SM bug

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Removes SUPERMATTER_IGNORE_1 flag
atom flag that is used in a small specific case is not good

## Why It's Good For The Game
snip
also bacon hates it

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

</details>

## Changelog
:cl:
code: removed SUPERMATTER_IGNORE_1 atom flag. SM has a static list that checks if it should not consume an atom.
fix: fixed where SM code causes runtime
/:cl:

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -40,7 +40,6 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define UNPAINTABLE_1 				(1<<17)
 #define HTML_USE_INITAL_ICON_1		(1<<18) 			//! Should we use the initial icon for display? Mostly used by overlay only objects
 /// Does the supermatter skip over this atom?
```
Comment (imsynthex) [inline code/__DEFINES/flags.dm]:
```suggestion

```

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -691,7 +702,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		user.dust(force = TRUE)
 		if(is_power_processing())
 			matter_power += 200
-	else if(AM.flags_1 & SUPERMATTER_IGNORES_1)
+	else if(consume_protected[AM.type])
```
Comment (itsmeow) [inline code/modules/power/supermatter/supermatter.dm]: typecaches use `isintypecache`.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -113,6 +113,9 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	///Can the crystal trigger the station wide anomaly spawn?
 	var/anomaly_event = TRUE
 
+	/// don't let these to be consumed by SM
+	var/static/list/consume_protected = list()
```
Comment (itsmeow) [inline code/modules/power/supermatter/supermatter.dm]: Better to have this be empty (null) since its gonna be replaced anyway

**Comment by EvilDragonfiend** [comment]
review addressed.
also, changed to check typecheck earlier so that it won't call consume() or not show unnecessary messages.

File: code/modules/power/supermatter/supermatter.dm [hunk 4]
```diff
@@ -113,6 +113,9 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	///Can the crystal trigger the station wide anomaly spawn?
 	var/anomaly_event = TRUE
 
+	/// don't let these to be consumed by SM
+	var/static/list/consume_protected
```
Comment (imsynthex) [inline code/modules/power/supermatter/supermatter.dm]: `not_dustable` would be a better, more descriptive name

**Review by imsynthex — CHANGES_REQUESTED** [review]

File: code/__DEFINES/flags.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/__DEFINES/flags.dm]: oh yeah I forgot it

File: code/modules/power/supermatter/supermatter.dm [hunk 5]
```diff
@@ -434,10 +445,11 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			air_update_turf()
 
 	for(var/mob/living/carbon/human/l in viewers(HALLUCINATION_RANGE(power), src)) // If they can see it without mesons on.  Bad on them.
-		if(!HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE) && !HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE))
-			var/D = sqrt(1 / max(1, get_dist(l, src)))
-			l.hallucination += power * config_hallucination_power * D
-			l.hallucination = CLAMP(0, 200, l.hallucination)
+		if(!HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE) || !HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE))
```
Comment (itsmeow) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
		if(HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE) || HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE))
```

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/power/supermatter/supermatter.dm (hunk 5 above)
Comment (EvilDragonfiend) [reply code/modules/power/supermatter/supermatter.dm]: Oh shit I missed it lol

**Review by itsmeow — APPROVED** [review]

**Pull request merged**